loopy/target/pyopencl.py

```diff
@@ -821,12 +889,210 @@ def get_kernel_call(self,
 # }}}
 
 
+# {{{ split_args_for_overflow
+
+def split_args_for_overflow(
+        kernel: LoopKernel, passed_names: Sequence[str],
+        *, limit_arg_size_nbytes: Optional[int], pointer_size_nbytes: int
+        ) -> Tuple[Sequence[str], Sequence[str]]:
+    if limit_arg_size_nbytes is None:
+        return passed_names, []
+
+    regular_arg_names = []
+    overflow_arg_names = []
+
+    # Consider that the pointer to the arg overflow struct also occupies
+    # argument space.
+    running_arg_size = pointer_size_nbytes
+
+    for arg_name in passed_names:
+        arg = kernel.get_var_descriptor(arg_name)
+        if isinstance(arg, (ValueArg, ArrayArg, ConstantArg, TemporaryVariable)):
+            if isinstance(arg, ValueArg):
+                arg_size = arg.dtype.numpy_dtype.itemsize
+            else:
+                arg_size = pointer_size_nbytes
+
+            if running_arg_size + arg_size > limit_arg_size_nbytes:
+                overflow_arg_names.append(arg_name)
+            else:
+                regular_arg_names.append(arg_name)
+
+            running_arg_size += arg_size
+
+        elif isinstance(arg, ImageArg):
+            regular_arg_names.append(arg_name)
```

Comment:
Should this also increment running_arg_size?

Reply:
As far as I understand, images, on most hardware of which I am aware, are passed via a different mechanism that does not consume parameter space, i.e. texture bindings. That said, there's nothing really keeping the runtime from consuming some par space per image anyway... but at that point we're in the realm of guesswork. So I'd say let's wait for it to break before speculating.